src/audio/mixer.c

```diff
@@ -217,6 +217,7 @@ static int mixer_params(struct comp_dev *dev,
 	return 0;
 }
 
+#if CONFIG_IPC_MAJOR_3
```

Comment:
Sorry @RanderWang - I'm not understanding this change. Why do we only comment out some trigger commands and not others. Can you split this patch into two parts for (1) and (2) above.

Reply:
these code will return PPL_STATUS_PATH_STOP to stop pipeline visit, then the components after mixer can't be started.  For current SOF pipeline, it will like    host-> mixer->dai, only one pipeline, but for ipc4, it will like
copier1 (host) -> mixin (pipeline 1) -> mixout -> copier2 (dai, pipeline 2). Now we use mixer to replace mixin & mixout but there are still two pipeline2. when mixer return PPL_STATUS_PATH_STOP since pipeline1 is active, then copier 2 will not be active.

Reply:
ok, got you - so I would do this differently. You already have comp_drivers for mixer, mixin and mixout so you can differentiate using the comp_driver ops. i.e.
```c
int trigger_common()
{
/* do trigger common for all mixer, mixin and mixout */
}

int mixin_trigger()
{
  /* do mixin specific stuff */
    trigger_common();
  /* do mixin specifc stuff */
}

struct comp_driver mixer = {
   .trigger = trigger_common,
};

struct comp_driver mixin = {
   .trigger = mixin_trigger,
};

```

Reply:
thanks, I get it.